### PR TITLE
util: Add Clang thread safety annotations for variables guarded by cs_args

### PR DESCRIPTION
**Pull request by practicalswift**
* Add missing `cs_args` locks
* Add Clang thread safety annotations for variables guarded by `cs_args`

### CODE REVIEW

**Comment by practicalswift** [comment]
Now includes a fix for a missing `cs_args` lock in #10267.

**Comment by practicalswift** [comment]
Rebased!

File: src/util.cpp [hunk 1]
```diff
@@ -848,7 +850,12 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
         }
         // if there is an -includeconf in the override args, but it is empty, that means the user
         // passed '-noincludeconf' on the command line, in which case we should not include anything
-        if (m_override_args.count("-includeconf") == 0) {
+        bool emptyIncludeConf;
+        {
+            LOCK(cs_args);
```
Comment (MarcoFalke) [inline src/util.cpp]: `ReadConfigFiles` is done once on start. Couldn't you just take the lock at the beginning of the method and remove those scoped locks further down in the method?

File: src/util.cpp [hunk 2]
```diff
@@ -387,6 +388,7 @@ void ArgsManager::WarnForSectionOnlyArgs()
     // if it's okay to use the default section for this network, don't worry
     if (m_network == CBaseChainParams::MAIN) return;
 
+    LOCK(cs_args);
```
Comment (MarcoFalke) [inline src/util.cpp]: Just take the lock in the very first line of this method, or is there a specific reason that `m_network` shouldn't be protected?

**Comment by practicalswift** [comment]
@MarcoFalke Thanks for reviewing! Feedback addressed. Please re-review :-)

**Comment by practicalswift** [comment]
Added `GUARDED_BY(cs_args)` to `m_available_args` and `m_network_only_args` too :-)

File: src/util.cpp [hunk 3]
```diff
@@ -304,6 +304,7 @@ class ArgsManagerHelper {
     static inline bool GetNetBoolArg(const ArgsManager &am, const std::string& net_arg)
     {
         std::pair<bool,std::string> found_result(false,std::string());
+        LOCK(am.cs_args);
```
Comment (MarcoFalke) [inline src/util.cpp]: I'd prefer to annotate this function and then take the lock where `GetNetBoolArg` is called (in `GetChainName`)

File: src/util.cpp [hunk 4]
```diff
@@ -886,7 +891,8 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
         }
         // if there is an -includeconf in the override args, but it is empty, that means the user
         // passed '-noincludeconf' on the command line, in which case we should not include anything
-        if (m_override_args.count("-includeconf") == 0) {
+        bool emptyIncludeConf = m_override_args.count("-includeconf") == 0;
+        if (emptyIncludeConf) {
```
Comment (MarcoFalke) [inline src/util.cpp]: Why this change?

**Comment by MarcoFalke** [comment]
utACK, but could squash into two commits (1) add missing locks and (2) add lock annotations

**Comment by MarcoFalke** [comment]
```
POTENTIAL DEADLOCK DETECTED
Previous lock order was:
 (1)cs_args  util.cpp:874 (2)csPathCached  util.cpp:767
Current lock order is:
 (2)csPathCached  util.cpp:767 (1)cs_args  util.cpp:508

**Comment by practicalswift** [comment]
@MarcoFalke Thanks! I'll investigate immediately!

**Comment by MarcoFalke** [comment]
Note that I reproduced this with `--enable-debug` in `./configure`

**Comment by practicalswift** [comment]
@MarcoFalke Strange! I always build with `--enable-debug`. Triggered by `make check`?

**Comment by MarcoFalke** [comment]
No, triggered by `bitcoind` or `bitcoin-qt`. It happens early on when the args and data dir are read.

File: src/util.cpp (hunk 1 above)
Comment (practicalswift) [reply src/util.cpp]: This change caused the deadlock. Now reverted to previous version :-)

**Comment by practicalswift** [comment]
@MarcoFalke Please re-review :-)

**Comment by practicalswift** [comment]
@MarcoFalke Reverted again :-) Please re-review!
